Adds a batch of cloud compute bug doc texts (#48329)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

[id="ocp-4-11-cloud-compute-bug-fixes"]
==== Cloud Compute

> * `CertificateSigningRequest` (CSR) resource renewal is handled by the Kubernetes controller manager and correctly left pending by the CLuster Machine Approver Operator, which increases the value of the `mapi_current_pending_csr` metric to `1`. Previously, when the  Kubernetes controller manager approved the CSR, the Operator ignored it and left the metric unchanged. As a result, the `mapi_current_pending_csr` metric was stuck at `1` until the Operator next reconciled. With this release, CSR approvals from other controllers are always reconciled to update metrics and the value of the `mapi_current_pending_csr` metric is updated after every reconcile. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2047702[*BZ#2047702*])

Reviewer: ```suggestion
* `CertificateSigningRequest` (CSR) resource renewal is handled by the Kubernetes controller manager and correctly left pending by the Cluster Machine Approver Operator, which increases the value of the `mapi_current_pending_csr` metric to `1`. Previously, when the  Kubernetes controller manager approved the CSR, the Operator ignored it and left the metric unchanged. As a result, the `mapi_current_pending_csr` metric was stuck at `1` until the Operator next reconciled. With this release, CSR approvals from other controllers are always reconciled to update metrics and the value of the `mapi_current_pending_csr` metric is updated after every reconcile. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2047702[*BZ#2047702*])
```

* Previously, only regions contained in a list of known regions within the AWS SDK were validated, and specifying any other region caused an error. This meant that, as new regions were added, they could not be used until the SDK was updated to contain the new region information. With this release, regions are validated with a less strict setting that warns the user when a region is not recognized. As a result, new regions might cause warning messages but can be used immediately. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2065510[*BZ#2065510*])

> * Previously, the Cluster Machine Approver Operator appended the `"Approved"` status condition to its condition list. As a result, the Kubernetes API server logged errors containing the message `[SHOULD NOT HAPPEN] failed to update managedFields`. With this release, the Operator is updated to check its conditions before appending to the list, only update the condition when necessary. As a result, the conditions are no longer duplicated in the `CertificateSigningRequest` resource and the Kubernetes API server no longer logs errors about the duplicate. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1978303[*BZ#1978303*])

Reviewer: ```suggestion
* Previously, the Cluster Machine Approver Operator appended the `"Approved"` status condition to its condition list. As a result, the Kubernetes API server logged errors containing the message `[SHOULD NOT HAPPEN] failed to update managedFields`. With this release, the Operator is updated to check its conditions before appending to the list, and only update the condition when necessary. As a result, the conditions are no longer duplicated in the `CertificateSigningRequest` resource and the Kubernetes API server no longer logs errors about the duplicate. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1978303[*BZ#1978303*])
```

* Previously, a defect in the Cisco ACI neutron implementation that was present in {rh-openstack-first} version 16, caused the query for subnets belonging to a given network to return unexpected results. As a result, the {rh-openstack} Cluster API provider might try to provision instances with duplicated ports on the same subnet, leading to a failed provisioning. With this release, additional filtering in the {rh-openstack} Cluster API provider ensures there is no more than one port per subnet, and it is now possible to deploy {product-title} on {rh-openstack} version 16 with Cisco ACI. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2033862[*BZ#2033862*])

* Previously, the {rh-openstack-first} Machine API provider did not use the proxy environment variable directives, causing installation behind an HTTP or HTTPS proxy to fail. With this release, the provider obeys proxy directives and functions correctly in a restricted environment in which egress traffic is only allowed through a proxy.
(link:https://bugzilla.redhat.com/show_bug.cgi?id=2046133[*BZ#2046133*])

> * Previously, when upgrading from OpenShift Container Version 4.9 to 4.10, inconsistencies between multiple controllers resulted in incorrect version numbers. As a result, the version number is not consistent. With this release, consistent reading of version numbers has been enacted and the release version is now stable in the cluster operator status. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2059716[*BZ#2059716*])

Reviewer: ```suggestion
* Previously, when upgrading from {product-title} 4.9 to 4.10, inconsistencies between multiple controllers resulted in incorrect version numbers. As a result, the version number was not consistent. With this release, consistent reading of version numbers has been enacted and the release version is now stable in the cluster operator status. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2059716[*BZ#2059716*])
```
